### PR TITLE
Add Euclidean GCD in Factor

### PR DESCRIPTION
**Pull request by nic-hartley**
The syntax is probably a little funky. Luckily, Factor has some pretty nice [learning resources](http://concatenative.org/wiki/view/Factor/Learning). Specifically, I'd recommend looking at [Your First Program](https://docs.factorcode.org/content/article-first-program.html), and having a decent grounding in functional programming; a lot of the concepts are similar.

And of course, if you have questions, comment on the lines. I'll do my best to answer.

### CODE REVIEW

**Comment by leios** [comment]
Oh, nice. Looks like you got everything. We'll review the code soon. Thanks a bunch!

**Comment by nic-hartley** [comment]
@leios Not yet. I've still got a few more algorithms to implement! (In separate PRs, of course, I read the contributing guide)

**Comment by leios** [comment]
Oh, I meant that this PR seemed to update everything for the inclusion of Factor into the AAA. Good job!

**Comment by jiegillet** [comment]
I call dibs on reviewing this one (and #426) if no one else wants it. Although I want to take my time and read a bit more about the language first.

File: contents/euclidean_algorithm/code/factor/euclid.factor [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ [
+  ! make sure the lower number is deeper
+  2dup >= [ swap ] when ! a b -> G(reater) L(ess)
+  over - ! leaves us with L G-L
+ ]
+ until
+ ! we have the GCD twice now, drop one
+ drop ;
+
+: euclid% ( a b -- gcd )
```
[the remark this answers is not in the document]
Comment (nic-hartley) [reply contents/euclidean_algorithm/code/factor/euclid.factor]: Whoops, yep, forgot. Also, didn't know about `bi@`, I've been using `<op> swap <op>`, but `[ <op> ] bi@` is better.

File: contents/euclidean_algorithm/code/factor/euclid.factor [hunk 2]
```diff
@@ -0,0 +1,31 @@
+: euclid- ( a b -- gcd )
```
[the remark this answers is not in the document]
Comment (nic-hartley) [reply contents/euclidean_algorithm/code/factor/euclid.factor]: ...I could have sworn I did. Sorry about that.

File: book.json [hunk 3]
```diff
@@ -145,6 +145,10 @@
           "name": "Fortran90"
         },
         {
+          "lang": "factor",
+          "name": "Factor"
+	},
```
[the remark this answers is not in the document]
Comment (nic-hartley) [reply book.json]: aaaAAAA

This is why no one likes Vim.

**Pull request merged**